### PR TITLE
feat: cookieless embed

### PR DESCRIPTION
**Pull request by bryans99**
Adds support for cookieless embed to the embed-sdk. This is an additive change and does not impact existing functionality. The actual change to the SDK is quite small. Note that cookieless embed does require the use of new endpoints. Examples of using these endpoints have been added to web development server.

The README has been updated to specify how to utilize cookieless embed.

The larger part of this change is to the demo application. The demo application has been substantially modified to make it easier to try out embed in various scenarios. A summary of the changes are as follows:
1. ids of objects to embed can now be specified in the .env file. this means it is no longer necessary to customize demo_config.ts.
2. how the embed application behaves can be changed through the demo UI. These changes are saved to local storage so survive a page reload.
3. The dashboard embed can be hidden/shown. This simulates how a customer might create additional IFRAMEs. 4, The look embed can be hidden/shown.
5. The explore embed can be hidden/shown.
6. The extension embed can be hidden/shown.
7. Fonts and components have been made a little prettier (not much but it is a little better).
8. The layout is a little reactive. On large monitors, the embeds may be displayed side by side.
9. Dashboards can be dynamically resized.
10. Dashboard navigation can be dynamically prevented allowed.
11. Cookieless embed can be dynamically turned on/off.
12. Status messages are now updated for more events.
13. A example of the using the embed postMessage API has been added. Note that this is NOT the embed SDK but was added to ensure the cookieless embed also works with postMessage.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@looker/embed-sdk",
-  "version": "1.6.2",
+  "version": "1.7.1-beta",
```
Comment (jkaster) [inline package.json:3]: Do you still want this as beta?

**Review by jkaster — APPROVED** [review]
Approved in case you intend to publish this as a beta release

File: package.json (hunk 1 above)
Comment (bryans99) [reply package.json:3]: For the moment it's fine but thanks for reminding me. I won't be merging this until the next release cut is done. At that point I will update this to 1.7.2(?) and publish. Publish is still manual

**Review by jkaster — DISMISSED** [review]
LGTM

**Review by jkaster — APPROVED** [review]

**Review by jkaster — DISMISSED** [review]
LGTM

**Review by jkaster — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -225,21 +225,21 @@ This process will be called every time a Looker embed IFRAME is created. The acq
...
 
-A time to live for each token is also returned for each token. It is important that the response of the `acquire_embed_cookieless_session` be returned to the browser with the exception of the `session_reference_token`.
+A time to live for each token is also returned. It is important that the response of the `acquire_embed_cookieless_session` be returned to the browser with the exception of the `session_reference_token`. The hosting application MUST keep track of the `session_reference_token` for each user.
 
-The example below uses javascript, but you may utilize the Looker SDK in the language of your choice. Note that this is very simplistic implementation for demonstration purposes only. An actual implementation should be a lot more robust.
+The example shown below is simplistic and uses an in memory cache to keep track of the `session_reference_token`. In memory caches will not work in clustered environments so a use a distributed cache such as `redis` in production. An alternative is to save the `session_reference_token` in an encrypted session cookie. The use of session cookies is demonstrated [here](/server_utils/routes.ts).
```
Comment (jkaster) [inline README.md:237]:
```suggestion
The example shown below is simplistic and uses an in memory cache to keep track of the `session_reference_token`. In memory caches will not work in clustered environments so use a distributed cache such as `redis` in production. An alternative is to save the `session_reference_token` in an encrypted session cookie. The use of session cookies is demonstrated [here](/server_utils/routes.ts).
```

**Review by jkaster — CHANGES_REQUESTED** [review]
Minor copy edit

File: README.md (hunk 2 above)
Comment (bryans99) [reply README.md:237]: I'm including this change in the next PR (against this branch) coming up soon.

**Review by jkaster — DISMISSED** [review]
LGTM

**Review by jkaster — APPROVED** [review]

File: server/utils/cookieless_utils.ts [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+
+// The Looker session
+let lookerSession: NodeSession
+
+/**
+ * Create a Looker session using the Looker SDK. This session can be
+ * be shared amongst all embed users. If data is required for the embed
+ * user, this session can be used to create a session for the embed user,
+ * something like the following but not required for this demo.
+ * <pre>
```
Comment (jkaster) [inline server/utils/cookieless_utils.ts:54]: not important but I think most doc systems support tripletick with syntax highlighting? Don't know if this doc generator does, though.

**Review by jkaster — CHANGES_REQUESTED** [review]
LGTM

File: server/utils/cookieless_utils.ts (hunk 3 above)
Comment (bryans99) [reply server/utils/cookieless_utils.ts]: Changed to use back ticks and re-ran doc generation. No change to output so it seems that it does

File: server/utils/cookieless_utils.ts [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+
+// The Looker session
+let lookerSession: NodeSession
+
+/**
+ * Create a Looker session using the Looker SDK. This session can be
+ * be shared amongst all embed users. If data is required for the embed
+ * user, this session can be used to create a session for the embed user,
+ * something like the following but not required for this demo.
+ * ```
```
Comment (jkaster) [inline server/utils/cookieless_utils.ts:54]: well, if you're going to go all this way add the `ts` or `js` language tag 😛

**Review by jkaster — DISMISSED** [review]
Looks mostly good to me

File: server/utils/cookieless_utils.ts (hunk 4 above)
Comment (bryans99) [reply server/utils/cookieless_utils.ts]: fixed

**Review by josephaxisa — APPROVED** [review]
lgtm

**Pull request merged**